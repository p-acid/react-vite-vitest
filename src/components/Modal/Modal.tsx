import { HTMLAttributes } from "react";

interface ModalProps extends HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
  onClose: () => void;
}

const Modal = ({ isOpen, onClose, children }: ModalProps) => {
  if (!isOpen) return null;

  return (
    <div className="modal" data-testid="modal">
      <div className="modal-content">
        {children}
        <button className="modal-close" onClick={onClose}>
          Close
        </button>
      </div>
    </div>
  );
};

export default Modal;
