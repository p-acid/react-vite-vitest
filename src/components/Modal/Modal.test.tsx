import { fireEvent, render } from "@testing-library/react";

import { describe, expect, test, vitest } from "vitest";
import Modal from "./Modal";

describe("Modal", () => {
  test("renders nothing when isOpen is false", () => {
    const { queryByTestId } = render(
      <Modal isOpen={false} onClose={() => {}}>
        Test Content
      </Modal>
    );
    expect(queryByTestId("modal")).toBeNull();
  });

  test("renders modal when isOpen is true", () => {
    const { getByTestId } = render(
      <Modal isOpen={true} onClose={() => {}}>
        Test Content
      </Modal>
    );
    expect(getByTestId("modal"));
  });

  test('calls onClose when "Close" button is clicked', () => {
    const onClose = vitest.fn();
    const { getByText } = render(
      <Modal isOpen={true} onClose={onClose}>
        Test Content
      </Modal>
    );
    fireEvent.click(getByText("Close"));
    expect(onClose).toHaveBeenCalled();
  });

  // Snapshot Test
  test("matches snapshot", () => {
    const { asFragment } = render(
      <Modal isOpen={true} onClose={() => {}}>
        Test Content
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
