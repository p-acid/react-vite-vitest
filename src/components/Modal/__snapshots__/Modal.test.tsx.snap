// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
    data-testid="modal"
  >
    <div
      class="modal-content"
    >
      Test Content
      <button
        class="modal-close"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
